### PR TITLE
Pin version of jackson-dataformat-xml transitive deps

### PR DESCRIPTION
**Pull request by ksobolew**
## Description

This is a transitive dependency of a transitive dependency, but it's there and is the only one of the Jackson family which is not pinned to the `dep.jackson.version` version.

## Additional context and related issues

Should be obsoleted by airlift/airbase#346

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by wendigo — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -1224,6 +1224,13 @@
                 <version>3.18.1</version>
             </dependency>
 
+            <dependency>
+                <!-- pinned for transitive dependencies; TODO: move this to airbase -->
+                <groupId>com.fasterxml.jackson.dataformat</groupId>
+                <artifactId>jackson-dataformat-xml</artifactId>
```
Comment (hashhar) [inline pom.xml:1230]: do we need this for every new transitive dependency we pull? Does jackson provide a bom that we can import?

**Review by hashhar — APPROVED** [review]
LGTM

some future proofing ideas, no changes requested

File: pom.xml (hunk 1 above)
Comment (ksobolew) [reply pom.xml:1230]: Actually, they do! But I would consider this to be temporary and doing the proper solution in Airbase - I'll update airlift/airbase#346 to use it. Thanks for pointing this out!

**Comment by ksobolew** [comment]
Obsoleted by #16971

**Pull request closed without merging**